entities/storobj/storage_object.go

```diff
@@ -1131,6 +1179,57 @@ func unmarshalTargetVectors(rw *byteops.ReadWriter) (map[string][]float32, error
 	return nil, nil
 }
 
+// unmarshalMultiVectors unmarshals the multi vectors from the buffer. If onlyUnmarshalNames is set and non-empty,
+func unmarshalMultiVectors(
+	rw *byteops.ReadWriter,
+	onlyUnmarshalNames map[string]interface{},
+) (map[string][][]float32, error) {
+	// This check prevents from panic when somebody is upgrading from version that
+	// didn't have multi vector support. This check is needed bc with the multi vectors
+	// feature the storage object can have vectors data appended at the end of the file
+	if rw.Position < uint64(len(rw.Buffer)) {
+		multiVectorsOffsets := rw.ReadBytesFromBufferWithUint32LengthIndicator()
+		multiVectorsSegmentLength := rw.ReadUint32()
+		pos := rw.Position
+
+		if len(multiVectorsOffsets) > 0 {
+			var mvOffsets map[string]uint32
+			if err := msgpack.Unmarshal(multiVectorsOffsets, &mvOffsets); err != nil {
+				return nil, fmt.Errorf("Could not unmarshal multi vectors offset: %w", err)
```

Comment:
Nit: Go says error strings [should not be capitalized](https://go.dev/wiki/CodeReviewComments#error-strings)

Nit-2: (hard as I tried I couldn't find the original blog post / code style guide) I remember it being conventional to have the error message say what the code was _trying to do_, not what it failed to do. That is, avoiding "couldn't" and "error while" and so on. But then Google's [styleguide](https://google.github.io/styleguide/go/best-practices#adding-information-to-errors) is full with those, so yeah 🤷